frontend/helpers/issues.ts

```diff
@@ -15,34 +15,50 @@
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import { get, omit } from 'lodash';
+import { get } from 'lodash';
 import { getAPIUrl } from '../services/api';
 import { STATUSES_COLOURS, STATUSES_ICONS, STATUSES } from '../constants/issues';
 import { isAdmin, hasPermissions, PERMISSIONS } from './permissions';
-import { sortByDate } from './sorting';
+
+const renameFieldIfExists = (issue, fieldName, newFieldName) => {
+	if (issue[fieldName]) {
+		issue[newFieldName] = issue[fieldName];
+	}
+};
 
 export const prepareIssue = (issue, jobs = []) => {
-	const thumbnail = getAPIUrl(issue.thumbnail);
-	const descriptionThumbnail = issue.viewpoint && issue.viewpoint.screenshot
+	if (issue.thumbnail) {
+		const thumbnail = getAPIUrl(issue.thumbnail);
+		issue = {...issue, thumbnail};
+	}
+
+	const descriptionThumbnail = get(issue, 'viewpoint.screenshot')
 		? getAPIUrl(issue.viewpoint.screenshot)
 		: (issue.descriptionThumbnail || '');
-	const { Icon, color } = this.getStatusIcon(issue.priority, issue.status);
-	const roleColor = get(jobs.find((job) => job.name === get(issue.assigned_roles, '[0]')), 'color');
-
-	return {
-		...issue,
-		defaultHidden: issue.status === STATUSES.CLOSED,
-		name: issue.name,
-		description: issue.desc,
-		author: issue.owner,
-		createdDate: issue.created,
-		thumbnail,
-		descriptionThumbnail,
-		StatusIconComponent: Icon,
-		statusColor: color,
-		roleColor,
-		comments: issue.comments || []
-	};
+
+	if (descriptionThumbnail) {
+		issue = {...issue, descriptionThumbnail};
+	}
+
+	if (issue.priority && issue.status) {
+		const { Icon, color } = this.getStatusIcon(issue.priority, issue.status);
+		issue = {...issue, StatusIconComponent: Icon, statusColor: color};
+	}
+
+	if (issue.assigned_roles) {
+		const roleColor = get(jobs.find((job) => job.name === get(issue.assigned_roles, '[0]')), 'color');
+		issue = {...issue, roleColor};
+	}
+
+	if (issue.status) {
+		issue = {...issue, defaultHidden: issue.status === STATUSES.CLOSED};
+	}
+
+	renameFieldIfExists(issue, 'desc', 'description');
```

Comment:
Isn't better to return proper field name from the backend? :)

Reply:
I think you guys renamed it when you migrated issues, so he was just sticking to the convention.

I do agree that it should just use the backend name though, but it does mean it has to be changed for the whole stack.

Should it be a separate issue?

Reply:
> I think you guys renamed it when you migrated issues, so he was just sticking to the convention.`
Yes, we did it, but @sanmont3drepo refactored this code. It was more like a suggestion. :)

> Should it be a separate issue?
👍

Reply:
@simplypixi   you authored that code:
https://github.com/3drepo/3drepo.io/blob/staging/frontend/helpers/issues.ts#L36

Reply:
ok Let's create a new issue for this(https://github.com/3drepo/3drepo.io/issues/1604#issue-464626586). Marking it as resolved here.

@simplypixi  maybe you guys can take a look after all the migration is done?

Marking this as resolved in here.

Reply:
@sanmont3drepo Yes, I know that. I'm sure I had some reason to do that, but I can't remember what it is - it's been while ago. :P